### DISCUSSION TITLE
Discussion Topics for Oct 18, 2021 Engineering Sync

### DISCUSSION THREAD

**Discussion by carlosfelix2** [CDCgov/prime-reportstream#2640] [Show and tell]
Discussion Topics for Oct 18, 2021 Engineering Sync

**Comment by whytheplatypus** [comment]
Talk through Guiding Questions and Implementation Principles and maybe come up with a name for this document.


**Guiding Questions**
_For every piece of work_
- Does this meet/improve code clarity?
	- Does this introduce any magic?
- Does this improve separation of concerns? (Does this thing have more than one reason to change?)
- Does this follow Dependency Inversion?


Implementation "Principles"
- Don't stop with your first draft.
- Keep it simple to make it faster.
- Don't patch bad code -- rewrite it.
- Write clearly -- don't sacrifice clarity for "efficiency".
- Note "bad code" if you have to write it.
- Any _large piece_ should have a design phase.
	- Anything pointed at 5 for larger we should ask if this needs a design phase. Or anything we need an epic for.

**Reply by whytheplatypus** [reply]
Is there an existing pattern for this in the project that could be used?

**Reply by whytheplatypus** [reply]
Enforced by more than one (at least two) people agreeing that these have been followed

**Comment by ronaldheft-gov** [comment]
Two announcements:
* Intro for the SemanticBits contractors who should be joining today
* Invite everyone on ReportStream to join the Azure / PRIME infrastructure sessions I'll be running for SemanticBits

**Comment by jimduff-usds** [comment]
- Need a process to own depandabot PRs.
